src/ripple/app/misc/impl/Transaction.cpp

```diff
@@ -132,6 +133,23 @@ Transaction::pointer Transaction::load(uint256 const& id, Application& app)
         soci::blob sociRawTxnBlob (*db);
         soci::indicator rti;
 
+        if(searchedAll)
+        {
+            uint64_t count = 0;
+
+            *db << "SELECT COUNT(DISTINCT LedgerSeq) FROM Transactions WHERE LedgerSeq BETWEEN "
+                << range.first ()
+                << " AND "
+                << range.last ()
+                << ";",
+                soci::into (count, rti);
+
+            if (!db->got_data () || rti != soci::i_ok)
+                return {};
```

Comment:
I don't think I'd return early here. Instead I'd set `searchedAll` to `false`

Reply:
You think its possible for us to have the next query succeed even if this one fails?

Reply:
@undertome I don't see a way for this to fail and the next one to succeed. I agree, code is fine as-is.